web-app/src/app/screens/PostRegistration.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import CssBaseline from '@mui/material/CssBaseline';
+import Box from '@mui/material/Box';
+import Container from '@mui/material/Container';
+import Typography from '@mui/material/Typography';
+import { EmailOutlined, InfoOutlined, Sync } from '@mui/icons-material';
+import { Alert, Button } from '@mui/material';
+import { useDispatch, useSelector } from 'react-redux';
+import { emailVerified, verifyEmail } from '../store/profile-reducer';
+import {
+  selectEmailVerificationError,
+  selectIsVerificationEmailSent,
+  selectUserProfileStatus,
+} from '../store/profile-selectors';
+import { type AppError } from '../types';
+import { app } from '../../firebase';
+import { useEffect } from 'react';
+import { ACCOUNT_TARGET } from '../constants/Navigation';
+import { useNavigate } from 'react-router-dom';
+export default function PostRegistration(): React.ReactElement {
+  const dispatch = useDispatch();
+  const navigateTo = useNavigate();
+  const selectResendEmailSuccess = useSelector(selectIsVerificationEmailSent);
+  const selectResendEmailError = useSelector(selectEmailVerificationError);
+  const userProfileStatus = useSelector(selectUserProfileStatus);
+  const [resendEmailSuccess, setResendEmailSuccess] = React.useState(false);
+  const [resendEmailError, setResendEmailError] =
+    React.useState<AppError | null>(null);
+  React.useEffect(() => {
+    setResendEmailSuccess(selectResendEmailSuccess);
+  }, [selectResendEmailSuccess]);
+  React.useEffect(() => {
+    setResendEmailError(selectResendEmailError ?? null);
+  }, [selectResendEmailError]);
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      void app
+        .auth()
+        .currentUser?.reload()
+        .then(() => {
+          const currentUser = app.auth().currentUser;
+
+          if (currentUser !== null && currentUser.emailVerified) {
+            dispatch(emailVerified());
+            clearInterval(intervalId);
+          }
+        });
+    }, 1000);
```

Comment:
[suggestion]: The would increase this interval to at least 5 seconds to reduce network usage, multiple calls can be simultaneous if the network is slow.